### PR TITLE
DEV 2132 use nexus and remove github urls

### PR DESCRIPTION
**Pull request by qqiao2024**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -7,37 +7,18 @@ stages:
...
-    - apt-get install -y postgresql-client-13
-    - pip install "tox<4"
-  needs: [pre-commit]
-
 tox:
   stage: test
   parallel:
     matrix:
-      - PY_VER: ['3.6', '3.7', '3.8']
+      - BUILD_PY_VERSION: [python3.6, python3.7, python3.8, python3.9]
```
Comment (stilesj-uchicago) [inline .gitlab-ci.yml:20]: Who is the 3.6 consumer?

**Review by stilesj-uchicago — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (qqiao2024) [reply .gitlab-ci.yml:20]: I will remove python3.6 in my next ticket.

**Review by BenX00 — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
             "pytest-cov[toml]",
             "sqlalchemy~=1.3.0",
             "sqlalchemy-utils>=0.32,<0.36.4",
-            "indexd @ git+ssh://git@github.com/NCI-GDC/indexd.git@2.11.1-rc.3#egg=indexd",
+            "indexd",
         ],
         "pytest_indexd": [
```
Comment (wrkhenddher) [inline setup.py:23]: What's the fundamental difference between "dev" and "pytest_indexd"?
I wonder if they can be merged into one?
(new ticket, new PR)

**Review by wrkhenddher — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (qqiao2024) [reply setup.py:23]: No, they can not. 
pytest_indexd is a test plugin to delegate database setup to pytest-postgresql and should be able to help us parallel our testing.

**Pull request merged**